### ISSUE TITLE
Credentials stored in plain text.

### ISSUE THREAD

**Issue by douglasg14b** [wallabag/wallabagger#164]
Given the commodity of privacy and security breaches these days, I would imagine this to be an extremely import requirement for the extension.

The lack of this is concerning, and begs the question if the same negligence for secret management is practiced in other services/software as well?

There is the requirement for a client Id and client Secret, why is this not enough to authenticate the client? The necessity for a Username and Password defeats the purpose of having client specific tokens, no?

**Comment by Rurik19** [comment]
Unfortunately, no. Refresh token lasts only two weeks (by default) and then it should be obtained again with user credentials. It is easy to make non-storing-credentials wallabagger, but who wants to enter credentials every two weeks?

**Comment by tcitworld** [comment]
Ref. https://github.com/wallabag/wallabag/issues/2800

**Comment by douglasg14b** [comment]
@Rurik19 

There are many ways to solve that problem (Such as indefinite tokens for the extension), storing user credentials in plaintext locally is not an acceptable method... It's negligent and cavalier handling of user secrets. It's exposing your users to unnecessary risk. 

This is not simply my opinion, this is an industry standard. There are minimum thresholds of sensitive data management to be considered non-negligent. 

There are countless, reputable, online resources that highlight this. Such as [OWASP](https://www.owasp.org/index.php/Top_10-2017_A3-Sensitive_Data_Exposure), [CWE](https://cwe.mitre.org/data/definitions/256.html), [Microsoft Blog](https://www.microsoft.com/security/blog/2012/01/16/secure-credential-storage/). As well as a multitude of blogs, posts, forum topics, questions....etc Like [Coding Horror](https://blog.codinghorror.com/youre-probably-storing-passwords-incorrectly/) or [ PIA Blog](https://www.privateinternetaccess.com/blog/2018/04/another-day-another-breach-at-what-point-does-storing-passwords-in-plaintext-become-criminally-negligent/). As well as informational posts on how to tackle or think about this problem, such as [this](https://crypto.stackexchange.com/questions/10086/storing-credentials-for-a-third-party), or [this](https://blog.envkey.com/managing-passwords-and-secrets-common-anti-patterns-2d5d2ab8e8ca), or [this](https://security.stackexchange.com/questions/7580/storing-the-third-party-credentials-in-the-database-some-secure-place). 

Entire libraries and standards have been created SPECIFICALLY for this problem, such as [OAuth](https://en.wikipedia.org/wiki/OAuth).

>who wants to enter credentials every two weeks?

This is not an excuse to expose user credentials and put your users at risk. It's a problem to solve.
_________________________________

I'm not sure how some other applications do it, but Raindrop.io as an example immediately connects to your account when the extension is installed if you're already logged in.

**Comment by mdirik** [comment]
This is duplicate of #145 and #151.

**Comment by vext01** [comment]
I must admit, I did find it odd that the extension needed a client key *and* my account credentials. It seems that storing the account credentials undermines the idea of the client keys.

Ideally, if an attacker compromises one device, the others are untouched, but since the attacker basically owns the account at that point, they could revoke other devices etc.

I like the way github does things with personal access tokens.

**Comment by anarcat** [comment]
> I must admit, I did find it odd that the extension needed a client key and my account credentials. It seems that storing the account credentials undermines the idea of the client keys.

Yes. It makes the client key absolutely pointless because they give no additional security: to revoke access to a client, you still need to rotate the main account password because it's everywhere.
